consensus/spos/bls/v2/subroundBlock.go

```diff
@@ -286,6 +288,27 @@ func (sr *subroundBlock) sendBlockHeader(
 	sr.SetData(headerHash)
 	sr.SetHeader(headerHandler)
 
+	if headerHandler.GetShardID() == common.MetachainShardId {
```

Comment:
this can be simplified if you create a function and call it in both places.
e.g.
```go
func logHeader(header data.HeaderHandler) {
    jsonMarshalizer := &marshal.JsonMarshalizer{}
    jsonBytes, err := jsonMarshalizer.Marshal(header)
    if err == nil {
	log.Debug("Proposed header sent", "header", string(jsonBytes))
    }   
}

```